test/csuite/timestamp_abort/main.c

```diff
@@ -1113,6 +1120,85 @@ print_missing(REPORT *r, const char *fname, const char *msg)
           r->exist_key, r->last_key, r->first_key, r->last_key);
 }
 
+/*
+ * backup_exists --
+ *     Check whether the backup with the given ID exists in the database.
+ */
+static bool
+backup_exists(WT_CONNECTION *conn, uint32_t index)
+{
+    WT_CURSOR *cursor;
+    WT_SESSION *session;
+    char backup_id[64];
+    const char *idstr;
+    bool found;
+
+    testutil_check(__wt_snprintf(backup_id, sizeof(backup_id), "ID%" PRIu32, index));
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    testutil_check(session->open_cursor(session, "backup:query_id", NULL, NULL, &cursor));
+    found = false;
+    while (cursor->next(cursor) == 0) {
+        testutil_check(cursor->get_key(cursor, &idstr));
+        if (strcmp(idstr, backup_id) == 0) {
+            found = true;
+            break;
+        }
+    }
+    testutil_check(cursor->close(cursor));
+
+    testutil_check(session->close(session, NULL));
+    return (found);
+}
+
+/*
+ * backup_verify --
+ *     Verify previous backups.
+ */
+static void
+backup_verify(WT_CONNECTION *conn)
+{
+    struct dirent *dir;
+    struct stat sb;
+    DIR *d;
+    size_t len;
+    uint32_t index;
+    char backup_id[64], buf[1024];
+
+    testutil_assert_errno((d = opendir(".")) != NULL);
+    len = strlen(BACKUP_BASE);
+    while ((dir = readdir(d)) != NULL) {
+        if (strncmp(dir->d_name, BACKUP_BASE, len) == 0) {
+
+            /* Verify the backup only if it has completed. */
+            testutil_check(__wt_snprintf(buf, sizeof(buf), "%s/done", dir->d_name));
+            if (stat(buf, &sb) != 0) {
+                testutil_assert_errno(errno == ENOENT);
+                continue;
+            }
+
+            index = (uint32_t)atoi(dir->d_name + len);
+
+            if (backup_verify_quick) {
```

Comment:
We should either keep the two modes of backup verification — `backup_verify_quick`, which uses `testutil_verify_src_backup`, and the other one, which opens the database and reuses `recover_and_verify` — and keep this `if` statement, or if we are not keeping it, then we should also remove this `if` statement and the corresponding declaration and uses of `backup_verify_quick`.

I'd prefer to keep the two options for verification (unless the other one is broken), as the one that reuses `recover_and_verify` should check most if not all of the keyspace, so it should detect if there are any inconsistencies as in the HELP ticket. Or perhaps we might at least want to keep it until we know that `testutil_verify_src_backup` works well for sure.

Reply:
Ok. Yes, after today's discussion, I think we should keep both modes of verfication.